src/Prefetch.cpp

```diff
@@ -0,0 +1,346 @@
+#include <algorithm>
+#include <map>
+#include <string>
+#include <limits>
+
+#include "Prefetch.h"
+#include "CodeGen_Internal.h"
+#include "IRMutator.h"
+#include "IROperator.h"
+#include "Bounds.h"
+#include "Scope.h"
+#include "Simplify.h"
+#include "Substitute.h"
+#include "Util.h"
+
+namespace Halide {
+namespace Internal {
+
+using std::map;
+using std::string;
+using std::vector;
+using std::stack;
+using std::pair;
+
+// Prefetch debug levels
+int dbg_prefetch1 = 1;
```

Comment:
These should (at a minimum) be declared as constexpr rather than writable global data.